test/pardot-test-srv/main.go

```diff
@@ -0,0 +1,256 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"math/rand/v2"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/letsencrypt/boulder/cmd"
+)
+
+type config struct {
+	// OAuthAddr is the address (e.g. IP:port) on which the OAuth server will
+	// listen.
+	OAuthAddr string
+
+	// PardotAddr is the address (e.g. IP:port) on which the Pardot server will
+	// listen.
+	PardotAddr string
+
+	// ExpectedClientID is the client ID that the server expects to receive in
+	// requests to the /services/oauth2/token endpoint.
+	ExpectedClientID string `validate:"required"`
+
+	// ExpectedClientSecret is the client secret that the server expects to
+	// receive in requests to the /services/oauth2/token endpoint.
+	ExpectedClientSecret string `validate:"required"`
+
+	// DevelopmentMode is a flag that indicates whether the server is running in
+	// development mode. In development mode, the server will store contacts and
```

Comment:
Why is this flag needed? This service is only ever used in development mode, right?

Reply:
I wrote about this in the PR description:

> Add pardot-test-srv, a test-only service for mocking communication with Salesforce OAuth and Pardot APIs in non-production environments. Since Salesforce does not provide Pardot functionality in developer sandboxes, pardot-test-srv must run in all non-production environments (e.g., sre-development and staging).

Reply:
I have removed this flag after adjusting the tool to use a slice capped at 20 contacts that removes the oldest entries to make room for new ones. This also necessitated adding authentication to the contacts endpoint.